cmd/root.go

```diff
@@ -66,34 +71,58 @@ var rootCmd = &cobra.Command{
 	},
 }
 
+// check if the version file exists in the container
+func checkVersionExists(versionUrl string) bool {
+	resp, err := http.Get(versionUrl)
+	if err != nil {
+		log.Err("checkVersionExists: error getting version file from container [%s]", err)
+		return false
+	}
+
+	return resp.StatusCode != 404
+}
+
+func getRemoteVersion(req string) (string, error) {
+	resp, err := http.Get(req)
+	if err != nil {
+		log.Err("getRemoteVersion: error listing version file from container [%s]", err)
+		return "", err
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Err("getRemoteVersion: error reading body of response [%s]", err)
+		return "", err
+	}
+
+	var versionList ContainerList
+	err = xml.Unmarshal(body, &versionList)
+	if err != nil {
+		log.Err("getRemoteVersion: error unmarshalling xml response [%s]", err)
+		return "", err
+	}
+
+	if len(versionList.Blobs) != 1 {
+		return "", fmt.Errorf("the latest version container is empty")
+	}
+
+	versionName := strings.Split(versionList.Blobs[0].Name, "/")[1]
```

Comment:
Should we add a warning? Do we expect there to be 1 here always?

Reply:
our pipeline will make sure to delete the older file and create a new file so ideally it shall be always one as we can have only one latest version of release. But just in case by any mistake if there exists multiple files we shall add an error log that we have failed to determine the latest version and shall not proceed with the checks as it might lead to invalid conclusions.

Reply:
Ideally there should only be one file for the latest version in the container.
But if there are more than one file, then we can add some error logs and use the logic that Vikas suggested

Reply:
Added the logic where it will check that if the number of files in the latest version container is not equal to 1, then it will return an error and also log that. Please check.